dspace-server-webapp/src/main/java/org/dspace/app/rest/signposting/controller/LinksetRestController.java

```diff
@@ -0,0 +1,174 @@
+    public ResponseEntity getOne() {
+        return ResponseEntity.status(HttpStatus.METHOD_NOT_ALLOWED).build();
+    }
+
+    @PreAuthorize("hasPermission(#uuid, 'ITEM', 'READ')")
+    @RequestMapping(value = "/linksets" + REGEX_REQUESTMAPPING_IDENTIFIER_AS_UUID + "/json",
+            method = RequestMethod.GET, produces = "application/linkset+json")
+    public LinksetRest getJson(HttpServletRequest request, @PathVariable UUID uuid) {
+        try {
+            Context context = ContextUtil.obtainContext(request);
+
+            Item dso = itemService.find(context, uuid);
+            if (dso == null) {
+                throw new ResourceNotFoundException("No such Item: " + uuid);
+            }
+
+            List<LinksetNode> linksetNodes = new ArrayList<>();
+            if (dso.getType() == Constants.ITEM) {
+                List<ItemSignpostingProcessor> ispp = new DSpace().getServiceManager()
+                        .getServicesByType(ItemSignpostingProcessor.class);
+                for (ItemSignpostingProcessor sp : ispp) {
+                    sp.addLinkSetNodes(context, request, dso, linksetNodes);
+                }
+            }
+            return converter.toRest(LinksetMapper.map(linksetNodes), utils.obtainProjection());
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @PreAuthorize("hasPermission(#uuid, 'ITEM', 'READ')")
+    @RequestMapping(value = "/linksets" + REGEX_REQUESTMAPPING_IDENTIFIER_AS_UUID,
+            method = RequestMethod.GET, produces = "application/linkset")
+    public LinksetRest getLset(HttpServletRequest request, @PathVariable UUID uuid) {
+        try {
+            Context context = ContextUtil.obtainContext(request);
+
+            Item item = itemService.find(context, uuid);
+            if (item == null) {
+                throw new ResourceNotFoundException("No such Item: " + uuid);
+            }
+
+            List<LinksetNode> linksetNodes = new ArrayList<>();
+            List<ItemSignpostingProcessor> ispp = new DSpace().getServiceManager()
+                    .getServicesByType(ItemSignpostingProcessor.class);
+            for (ItemSignpostingProcessor sp : ispp) {
+                sp.addLinkSetNodes(context, request, item, linksetNodes);
+            }
+
+            LinksetRest linksetRest = null;
+            for (LinksetNode linksetNode : linksetNodes) {
+                if (linksetRest == null) {
+                    linksetRest = converter.toRest(linksetNode, utils.obtainProjection());
+                } else {
+                    linksetRest.getLinksetNodes().add(linksetNode);
+                }
+            }
+            return linksetRest;
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @PreAuthorize("hasPermission(#uuid, 'ITEM', 'READ') && hasPermission(#uuid, 'BITSTREAM', 'READ')")
```

Comment:
Yes, you are right, object cannot be `Item` **and** `Bitstream` at the same time. But in case of `hasPermission()` [current implementation](https://github.com/DSpace/DSpace/blob/7bbb5f438ba70d9250f707921b1f4ab036d45c11/dspace-server-webapp/src/main/java/org/dspace/app/rest/security/AuthorizeServicePermissionEvaluatorPlugin.java#L84) one of `hasPermission(#uuid, 'ITEM', 'READ')` `hasPermission(#uuid, 'BITSTREAM', 'READ')` will always be `true` because it can be ITEM and BITSTREAM at the same time. So using and(&&) is the right way

Reply:
Thank you for the explanation and the addition of the inline comment to describe it.  The code *definitely* looks confusing, but it makes sense now that I look closer.